runtime/runtime/src/lib.rs

```diff
@@ -1482,23 +1483,53 @@ impl Runtime {
                 None
             };
 
+        // Retrieve delayed receipts to initiate prefetching during local receipt processing.
+        // TODO add time to delayed receipt processing metric.
```

Comment:
why don't we do this at the beginning of `apply`?

Reply:
The processing order is `transactions -> local_receipts -> delayed_receipts` where local receipts become available only as transactions are processed. The prefetcher's work queue is FIFO which implies prefetching should happen in the same order as processing. If delayed receipts were prefetched out of order (e.g. at the beginning), it could get in the way of prefetching for local receipts.

Making the prefetcher have a work queue with priorities [was brought up](https://near.zulipchat.com/#narrow/stream/295306-contract-runtime/topic/prefetcher.20priority.20queue/near/432465885), but it's a bigger change.